### PR TITLE
Suppress focal-specific warnings

### PR DESCRIPTION
**Pull request by mjcarroll**
Addresses #157 and fixes #150.

Both homebrew and Ubuntu Focal package a `cppzmq` version that is reported as 4.7.0, but the deprecation warnings and available functions are different.  I believe this is due in part to there not being an official 4.7.0 release.

This simply detects if we are building on Ubuntu Focal and adjusts accordingly.

Signed-off-by: Michael Carroll <michael@openrobotics.org>

### CODE REVIEW

File: src/Node.cc [hunk 1]
```diff
@@ -619,7 +619,7 @@ bool Node::Unsubscribe(const std::string &_topic)
   if (!this->dataPtr->shared->localSubscribers
       .HasSubscriber(fullyQualifiedTopic))
   {
-#if (CPPZMQ_VERSION >= 40700)
+#ifdef IGN_CPPZMQ_POST_4_7_0
```
Comment (chapulina) [inline src/Node.cc:622]: Homebrew is giving this warning:

`'setsockopt' is deprecated: from 4.7.0, use `set` taking option from zmq::sockopt`

Comment (mjcarroll) [reply src/Node.cc:622]: I'm a bit stumped why this is triggering a warning in `Node.cc`, but not in `NodeShared.cc`, where the same pattern is used.

File: CMakeLists.txt [hunk 2]
```diff
@@ -46,6 +46,19 @@ ign_find_package(IgnProtobuf
 # Find ZeroMQ
 ign_find_package(ZeroMQ VERSION 4 REQUIRED PRIVATE)
 
+if (CMAKE_SYSTEM_NAME MATCHES "Linux")
+  execute_process(COMMAND lsb_release -cs
+    OUTPUT_VARIABLE RELEASE_CODENAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_QUIET)
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply CMakeLists.txt]: Added a warning in 0fbc40f

File: src/Node.cc (hunk 1 above)
Comment (mjcarroll) [reply src/Node.cc:622]: I was including the incorrect header for the version check, this should be addressed now.

**Comment by chapulina** [comment]
The `Log_TEST` failed on Focal, not sure if it could be caused by this PR.

**Comment by mjcarroll** [comment]
> The Log_TEST failed on Focal, not sure if it could be caused by this PR.

It happens locally for me, as well.  I think because this test doesn't involve zmq, it's more likely a difference between bionic and focal library versions (possibly `sqlite`?)

https://github.com/ignitionrobotics/ign-transport/blob/ign-transport8/log/src/Log_TEST.cc#L250-L259

**Comment by nkoenig** [comment]
> > The Log_TEST failed on Focal, not sure if it could be caused by this PR.
> 
> It happens locally for me, as well. I think because this test doesn't involve zmq, it's more likely a difference between bionic and focal library versions (possibly `sqlite`?)
> 
> https://github.com/ignitionrobotics/ign-transport/blob/ign-transport8/log/src/Log_TEST.cc#L250-L259

Focal outputs a value of 7254000000ns. I'll try adding an #ifdef.

**Comment by caguero** [comment]
> > > The Log_TEST failed on Focal, not sure if it could be caused by this PR.
> > 
> > 
> > It happens locally for me, as well. I think because this test doesn't involve zmq, it's more likely a difference between bionic and focal library versions (possibly `sqlite`?)
> > https://github.com/ignitionrobotics/ign-transport/blob/ign-transport8/log/src/Log_TEST.cc#L250-L259
> 
> Focal outputs a value of 7254000000ns. I'll try adding an #ifdef.

I relaxed the expectation for not having to go back to this test (hopefully) after another Ubuntu release.

**Review by caguero — APPROVED** [review]

**Pull request merged**